### PR TITLE
[DOC] Rework C++ concept tutorial.

### PR DESCRIPTION
**Pull request by smehringer**
I replaced the argument parser example within the `c++ concept tutorial` with a different one, since argument parsing is not part of seqan3 anymore.

### CODE REVIEW

File: doc/tutorial/03_concepts/model_a_concept.cpp [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+// concept fooger:
+template <typename T>
+concept fooger = has_foo<T> && std::same_as<typename T::FOO, int>;
+//![concept]
+
+struct my_type
+{
+    using FOO = int;
+    char foo; // foo can be of any type, here it is of type `char`
```
Comment (eseiler) [inline doc/tutorial/03_concepts/model_a_concept.cpp:21]:
```suggestion
    char foo{}; // foo can be of any type, here it is of type `char`
```

File: doc/tutorial/03_concepts/model_a_concept.cpp [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+};
+
+//![main]
+int main()
+{
+    seqan3::debug_stream << fooger<my_type> << std::endl; // should print 1
```
Comment (eseiler) [inline doc/tutorial/03_concepts/model_a_concept.cpp:27]:
```suggestion
    seqan3::debug_stream << std::boolalpha << fooger<my_type> << '\n'; // should print true
```

`std::boolalpha` should work with our debug_stream?
Not sure if we want it :)

Comment (smehringer) [reply doc/tutorial/03_concepts/model_a_concept.cpp:27]: Although the output is nicer I think it makes the snippet more complex than it needs to be

File: doc/tutorial/03_concepts/index.md [hunk 3]
```diff
@@ -222,94 +224,91 @@ to check the "return type" of the transformation trait.
...
-```cpp
-struct custom_validator
-{
-    // ...
-};
+You can learn how to make your own alphabet model the SeqAn requirements in \ref howto_write_an_alphabet
 
-static_assert(seqan3::validator<custom_validator>);
-```
+In order to understand what "make a type model a concept" means in practical terms now, let's look at an easier
```
Comment (SGSSGene) [inline doc/tutorial/03_concepts/index.md:279]:
```suggestion
In order to understand what "make a type model a concept" means in practical terms, let's look at an easier
```

File: doc/tutorial/03_concepts/index.md [hunk 4]
```diff
@@ -222,94 +224,91 @@ to check the "return type" of the transformation trait.
...
+example in the next section.
 
-To formally satisfy the requirements, your functions don't need the correct behaviour, yet.
-Only the signatures need to be fully specified.
+# Formally satisfying the requirements of a concept
```
Comment (SGSSGene) [inline doc/tutorial/03_concepts/index.md:282]:
```suggestion
# Satisfying a concept
```

**Review by SGSSGene — COMMENTED** [review]
I noted a few things, to keep my attention. They are just some thoughts, most of them don't have to be integrated.

**Review by SGSSGene — APPROVED** [review]
Solid improvements, nice work!

**Pull request merged**